{{cookiecutter.app_name}}/app/src/main/java/{{cookiecutter.package_dir}}/util/SingleLiveEvent.kt

```diff
@@ -0,0 +1,36 @@
+package {{cookiecutter.package_name}}.util
+
+import androidx.lifecycle.LifecycleOwner
+import androidx.lifecycle.MutableLiveData
+import androidx.lifecycle.Observer
+import java.util.concurrent.atomic.AtomicBoolean
+
+/**
+ * Description: Custom mutable live data that used for single event
+ *              such as navigation (for configuration change), show toast..
+ */
+class SingleLiveEvent<T> : MutableLiveData<T>() {
```

Comment:
@canhtv-0838 
class này mục đích làm gì thế e

Reply:
Description: Custom mutable live data that used for single event
 *              such as navigation (for configuration change), show toast.. 
e định nghĩa sẵn cái này thôi a, giả dụ trong trường hợp bị observe nhiều lần ấy ạ, thì mình chỉ lắng nghe 1 lần. Cái này phù hợp sử dụng trong trường hợp show dialog, message ... để tránh bị nảy nhiều lần :>

Reply:
@canhtv-

Reply:
cứ để vậy đi a, nếu cần dùng thì gọi vì trước e gặp case này rồi. Rất hữu dụng trong việc xử lý dialog. Nếu không dùng Single thì vẫn dùng Mutable bình thường thôi.